core/object/class_db.cpp

```diff
@@ -1277,6 +1276,29 @@ void ClassDB::get_enum_constants(const StringName &p_class, const StringName &p_
 	}
 }
 
+PackedStringArray ClassDB::get_enum_display_names(const StringName &p_class, const StringName &p_enum) {
+	Locker::Lock lock(Locker::STATE_READ);
+	PackedStringArray ret;
+
+	ClassInfo *type = classes.getptr(p_class);
+	while (type) {
+		const ClassInfo::EnumInfo *constants = type->enum_map.getptr(p_enum);
+		if (constants) {
+			ret.resize(constants->display_names.size());
+			String *ret_write = ret.ptrw();
+
+			int i = 0;
+			for (const String &display_name : constants->display_names) {
+				ret_write[i] = display_name;
```

Comment:
It probably should auto add `:value` to each of the names to ensure it's order independent.